test/suites/integration/target_group_policy_test.go

```diff
@@ -0,0 +1,130 @@
+package integration
+
+import (
+	"time"
+
+	"github.com/aws/aws-application-networking-k8s/pkg/apis/applicationnetworking/v1alpha1"
+	"github.com/aws/aws-application-networking-k8s/test/pkg/test"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+var _ = Describe("Target Group Policy Tests", func() {
+	var (
+		deployment *appsv1.Deployment
+		service    *v1.Service
+		httpRoute  *gwv1beta1.HTTPRoute
+		policy     *v1alpha1.TargetGroupPolicy
+	)
+
+	BeforeEach(func() {
+		deployment, service = testFramework.NewNginxApp(test.ElasticSearchOptions{
+			Name:      "target-group-policy-test",
+			Namespace: k8snamespace,
+		})
+	})
+
+	AfterEach(func() {
+		testFramework.ExpectDeleted(
+			ctx,
+			deployment,
+			service,
+			policy,
+			httpRoute,
+		)
+	})
+
+	It("Update Protocol creates new Target Group", func() {
+		policy = testFramework.CreateTargetGroupPolicy(service, &test.TargetGroupPolicyConfig{
+			PolicyName: "test-policy",
+			Protocol:   aws.String(vpclattice.TargetGroupProtocolHttp),
+		})
+
+		httpRoute = testFramework.NewHttpRoute(testGateway, service, service.Kind)
+
+		testFramework.ExpectCreated(ctx, deployment, service, policy, httpRoute)
+
+		tg := testFramework.GetTargetGroup(ctx, service)
+
+		Expect(*tg.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttp))
+
+		testFramework.UpdateTargetGroupPolicy(policy, &test.TargetGroupPolicyConfig{
+			Protocol: aws.String(vpclattice.TargetGroupProtocolHttps),
+		})
+
+		testFramework.ExpectUpdated(ctx, policy)
+
+		httpsTG := testFramework.GetTargetGroupWithProtocol(ctx, service, "https", "http1")
+
+		Expect(*httpsTG.Protocol).To(Equal(vpclattice.TargetGroupProtocolHttps))
+	})
+
+	It("Delete Target Group Policy reset health check config for HTTP and HTTP1 Target Group", func() {
```

Comment:
That can be a better name : "Delete Target Group Policy will reset health check config to default value" ?